sympy/matrices/expressions/determinant.py

```diff
@@ -50,7 +51,10 @@ def det(matexpr):
     1
     """
 
-    return Determinant(matexpr).doit()
+    if isinstance(matexpr, MatrixExpr) and not method:
```

Comment:
The "and not method" condition should be "and method is not None" which is different.